### PR TITLE
blurb: get BPO# from branch, add commit template - update the PR

### PR DESCRIPTION
**Pull request by matrixise**
Update of the [pull request](https://github.com/python/core-workflow/pull/184) of @tiran 

Co-authored-by: Christian Heimes <christian@python.org>

### CODE REVIEW

**Comment by matrixise** [comment]
Updated and rebased with the master branch.

**Comment by matrixise** [comment]
Thanks @Mariatta

File: blurb/blurb.py [hunk 1]
```diff
@@ -913,24 +913,12 @@ def add():
...
-            with_space = "\n" + bpo_line + " \n"
-            if without_space not in text:
-                sys.exit("Can't find BPO line to ensure there's a space on the end!")
-            text = text.replace(without_space, with_space)
-            file.write(text)
+    # See if the branch name incidates the BPO number
+    branch_bpo, branch_suffix = get_bpo_git_branch()
 
-    init_tmp_with_template()
+    with open(tmp_path, "wt", encoding="utf-8") as f:
```
Comment (webknjaz) [inline blurb/blurb.py]: Could be
```suggestion
    pathlib.Path(tmp_path).write_text(template.format(bpo=branch_bpo))
```

**Comment by larryhastings** [comment]
So, in general I don't like this approach.  I propose an alternate approach, which you can see in #312 .  In short: add "--bpo" and "--section" flags to "blurb add", letting you specify those fields on the command-line.  This will let @tiran automate the problem the way he likes, and presumably @matrixise and everybody else too.  If #312 is accepted I plan to close this PR without accepting it.

**Comment by matrixise** [comment]
:+1: for the #312

I am fine if you close this PR and thank you for the other solution.

**Comment by hugovk** [comment]
Closing this in favour of the other solution at #312.

**Pull request closed without merging**